File: plsar-examples/authentication-authorization/src/main/kotlin/foo/AuthAccess.java
package foo;

import shape.model.User;
import shape.repo.UserRepo;
import perched.support.access.Accessor;
import qio.annotate.Element;
import qio.annotate.Inject;

import java.util.Set;

@Element
public class AuthAccess implements DbAccess {

    @Inject
    UserRepo userRepo;

    public String getPassword(String phone){
        User user = userRepo.getPhone(phone);
        if(user != null){
            return user.getPassword();
        }
        return "";
    }

    public Set<String> getRoles(String phone){
        User user = userRepo.getPhone(phone);
        Set<String> roles = userRepo.getUserRoles(user.getId());
        return roles;
    }

    public Set<String> getPermissions(String phone){
        User user = userRepo.getPhone(phone);
        Set<String> permissions = userRepo.getUserPermissions(user.getId());
        return permissions;
    }

}

File: plsar-examples/authentication-authorization/src/main/kotlin/foo/support/AuthAccess.kt
package foo.support

import `plsar-auth`.support.DbAccess
import plsar.annotate.Element
import plsar.annotate.Inject
import foo.model.User
import foo.repo.UserRepo

@Element
class AuthAccess : DbAccess {

    @Inject
    var userRepo: UserRepo? = null

    override fun getPassword(credential: String?): String? {
        var user: User? = userRepo?.getPhone(credential)
        if(user == null){
            user = userRepo?.getEmail(credential)
        }
        return user.password
    }

    override fun getRoles(credential: String?): Set<String> {
        var user: User = userRepo?.getPhone(credential)
        if(user == null){
            user = userRepo.getEmail(credential)
        }
        return userRepo.getUserRoles(user.id)
    }

    override fun getPermissions(credential: String?): Set<String> {
        var user: User = userRepo?.getPhone(credential)
        if(user == null){
            user = userRepo.getEmail(credential)
        }
        return userRepo.getUserPermissions(user.id)
    }
}